File: lib/src/main/java/com/qwwuyu/lib/utils/FileUtils.java
package com.qwwuyu.lib.utils;

import java.io.File;

public class FileUtils {
    /* ======================== other ======================== */
    public static File webInfFile(String path, String name) {
        String classesPath = FileUtils.class.getClassLoader().getResource("/").getPath();
        File file = new File(new File(new File(classesPath).getParentFile(), path), name);
        if (!file.getParentFile().exists()) {
            file.getParentFile().mkdirs();
        }
        return file;
    }

    /* ======================== util ======================== */
    public static boolean isChild(File parent, File child) throws Exception {
        File parentFile = parent.getCanonicalFile();
        File childFile = child.getCanonicalFile();
        if (parentFile.getParent() == null && childFile.getParent() == null) {
            return parentFile.getPath().equals(childFile.getPath());
        } else if (parentFile.getParent() == null) {
            return childFile.getPath().startsWith(parentFile.getPath());
        } else if (parentFile.getParent().equals(childFile.getParent())) {
            return parentFile.getPath().equals(childFile.getPath());
        } else {
            return childFile.getPath().startsWith(parentFile.getPath());
        }
    }

    public static boolean isFile(File file) {
        return exists(file) && file.isFile();
    }

    public static boolean isDirectory(File file) {
        return exists(file) && file.isDirectory();
    }

    public static boolean exists(File file) {
        return file != null && file.exists();
    }

    public static String getFileSize(long size) {
        if (size < 1024L) {
            return size + "B";
        } else if (size < 1024L * 1024) {
            return String.format("%.2fKB", size / 1024f);
        } else if (size < 1024L * 1024 * 1024) {
            return String.format("%.2fMB", size / 1024 / 1024f);
        } else {
            return String.format("%.2fGB", size / 1024 / 1024 / 1024f);
        }
    }
}

File: lib/src/main/kotlin/com/qwwuyu/lib/utils/FileUtils.kt
package com.qwwuyu.lib.utils

import java.io.File
import java.lang.Exception

object FileUtils {
    /* ======================== other ======================== */
    fun webInfFile(path: String, name: String): File {
        val classesPath = FileUtils::class.java.classLoader.getResource("/").path
        val file = File(File(File(classesPath).parentFile, path), name)
        if (!file.parentFile.exists()) {
            file.parentFile.mkdirs()
        }
        return file
    }

    /* ======================== util ======================== */
    @Throws(Exception::class)
    fun isChild(parent: File, child: File): Boolean {
        val parentFile = parent.canonicalFile
        val childFile = child.canonicalFile
        return if (parentFile.parent == null && childFile.parent == null) {
            parentFile.path == childFile.path
        } else if (parentFile.parent == null) {
            childFile.path.startsWith(parentFile.path)
        } else if (parentFile.parent == childFile.parent) {
            parentFile.path == childFile.path
        } else {
            childFile.path.startsWith(parentFile.path)
        }
    }

    fun delChildFile(file: File) {
        if (!file.isDirectory) return
        file.listFiles()?.let { files ->
            files.forEach { file ->
                file.delete()
            }
        }
    }

    fun isFile(file: File): Boolean {
        return exists(file) && file.isFile
    }

    fun isDirectory(file: File): Boolean {
        return exists(file) && file.isDirectory
    }

    @JvmStatic
    fun exists(file: File?): Boolean {
        return file != null && file.exists()
    }

    fun getFileSize(size: Long): String {
        return when {
            size < 1024L -> size.toString() + "B"
            size < 1024L * 1024 -> String.format("%.2fKB", size / 1024f)
            size < 1024L * 1024 * 1024 -> String.format("%.2fMB", size / 1024 / 1024f)
            else -> String.format("%.2fGB", size / 1024 / 1024 / 1024f)
        }
    }
}